packages/react/src/VideoPlayer/VideoPlayer.tsx

```diff
@@ -14,195 +13,95 @@ import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/compone
 
-  const handleOnPlay = useCallback(
-    e => {
-      setPlaying(true)
-      onPlay && onPlay(e)
-    },
-    [onPlay],
-  )
-
-  const handleOnPause = useCallback(
-    e => {
-      setPlaying(false)
-      onPause && onPause(e)
-    },
-    [onPause],
-  )
-
-  const handleOnPlaying = useCallback(
-    e => {
-      videoRef.current?.textTracks[0].cues && setTrackInformation(videoRef.current.textTracks[0].cues)
-      onPlaying && onPlaying(e)
-    },
-    [onPlaying, videoRef],
-  )
-
-  const handleOnLoadedMetadata = useCallback(
-    e => {
-      setTotalTime(videoRef.current?.duration || 0)
-      onLoadedMetadata && onLoadedMetadata(e)
-    },
-    [onLoadedMetadata, videoRef],
-  )
-
-  const handleOnTimeUpdate = useCallback(
-    e => {
-      setPlayedTime(videoRef.current?.currentTime || 0)
-      onTimeUpdate && onTimeUpdate(e)
-    },
-    [onTimeUpdate, videoRef],
-  )
-
-  const handleVideoPlayback = () => {
-    if (videoRef.current) {
-      if (videoRef.current.paused) {
-        videoRef.current.play()
-      } else {
-        videoRef.current.pause()
-      }
-    }
-  }
-
-  useEffect(() => {
-    if (videoRef.current) {
-      for (let i = 0; i < videoRef.current.textTracks.length; i++) {
-        videoRef.current.textTracks[0].mode = 'hidden'
-      }
-    }
-  }, [videoRef])
-
-  useEffect(() => {
-    const handleResize = () => {
-      const breakpoint = videoWrapperRef.current?.getBoundingClientRect().width
-      if (breakpoint && breakpoint < 650) {
-        videoWrapperRef.current.classList.add(styles['VideoPlayer__container--small'])
-      } else {
-        videoWrapperRef.current?.classList.remove(styles['VideoPlayer__container--small'])
-      }
-    }
-
-    const resizeObserver = new ResizeObserver(_ => {
-      if (videoWrapperRef.current) {
-        handleResize()
-      }
-    })
-
-    handleResize()
-
-    const currentRef = videoWrapperRef.current
-
-    resizeObserver.observe(videoWrapperRef.current as Element)
-    return () => {
-      resizeObserver.unobserve(currentRef as Element)
-    }
-  }, [videoWrapperRef, playing])
+  useVideoKeypressHandlers(videoWrapperRef)
 
   return (
     <div
       className={clsx(
         styles.VideoPlayer__container,
         styles.VideoPlayer__overlays,
-        !playing && styles.VideoPlayer__showOverlays,
+        !isPlaying && styles.VideoPlayer__showOverlays,
       )}
       ref={videoWrapperRef}
     >
-      <video
-        ref={videoRef}
-        title={title}
-        controls={false}
-        poster={poster}
-        className={clsx(styles.VideoPlayer, className)}
-        onPlay={handleOnPlay}
-        onPause={handleOnPause}
-        onPlaying={handleOnPlaying}
-        onLoadedMetadata={handleOnLoadedMetadata}
-        onTimeUpdate={handleOnTimeUpdate}
-        {...rest}
-      >
+      {/* eslint-disable-next-line jsx-a11y/media-has-caption */}
```

Comment:
This will need to be removed as there's an externally-managed static analysis audit that runs on this repo to check if we've disabled any `jsx-a11y` rules.

Reply:
Good point.

I can add `<track kind="captions" />` back in, but it's a bit annoying as it's completely redundant.

Really, we need a linter rule to make sure that you can't use `VideoPlayer` without passing a captions track instead, or we iterate through the children each render and yell at the user if they don't pass a track 😅 (I'm not really advocating for that, don't worry)

Reply:
Resolved in 3d65fee00e176151ec35c579d2bfe1f6a2ec3e31